Classes/AbstractDataHandlerListener.php

```diff
@@ -138,6 +171,142 @@ protected function getAllChangeSetValuesMatch($triggerConfiguration, $changedFie
     }
 
     /**
+     * Retrieves the name of the  Indexing Queue Configuration for a record assumes
+     * that $this->solrConfiguration has been initialized properly
+     *
+     * @param string $recordTable Table to read from
+     * @param int $recordUid Id of the record
+     * @param TypoScriptConfiguration $solrConfiguration
+     * @return string Name of indexing configuration
+     */
+    protected function getIndexingConfigurationName(
+        $recordTable,
+        $recordUid,
+        TypoScriptConfiguration $solrConfiguration
+    ) {
+        $name = $recordTable;
+        $indexingConfigurations = $solrConfiguration->getEnabledIndexQueueConfigurationNames();
+        foreach ($indexingConfigurations as $indexingConfigurationName) {
+            if (!$solrConfiguration->getIndexQueueConfigurationIsEnabled($indexingConfigurationName)) {
+                // ignore disabled indexing configurations
+                continue;
+            }
+
+            $record = $this->getRecordWhenIndexConfigurationIsValid($recordTable, $recordUid,
+                $indexingConfigurationName, $solrConfiguration);
+            if (!empty($record)) {
+                $name = $indexingConfigurationName;
+                // FIXME currently returns after the first configuration match
+                break;
+            }
+        }
+
+        return $name;
+    }
+
+    /**
+     * Retrieves a record, taking into account the additionalWhereClauses of the
+     * Indexing Queue configurations.
+     *
+     * @param string $recordTable Table to read from
+     * @param int $recordUid Id of the record
+     * @param TypoScriptConfiguration $solrConfiguration
+     * @return array Record if found, otherwise empty array
+     */
+    protected function getRecord($recordTable, $recordUid, TypoScriptConfiguration $solrConfiguration)
+    {
+        $record = [];
+        $indexingConfigurations = $solrConfiguration->getEnabledIndexQueueConfigurationNames();
+
+        foreach ($indexingConfigurations as $indexingConfigurationName) {
+            $record = $this->getRecordWhenIndexConfigurationIsValid($recordTable, $recordUid,
+                $indexingConfigurationName, $solrConfiguration);
+            if (!empty($record)) {
+                // if we found a record which matches the conditions, we can continue
+                break;
+            }
+        }
+
+        return $record;
+    }
+
+    /**
+     * This method return the record array if the table is valid for this indexingConfiguration.
+     * Otherwise an empty array will be returned.
+     *
+     * @param string $recordTable
+     * @param integer $recordUid
+     * @param string $indexingConfigurationName
+     * @param TypoScriptConfiguration $solrConfiguration
+     * @return array
+     */
+    private function getRecordWhenIndexConfigurationIsValid(
```

Comment:
Only make a function `private` if you absolutely need to / if it should not be touched by other developers. Otherwise it is common practice in TYPO3 to use `protected` so that other dev can change them using XCLASSes.

Also while at it, can you please fix the name, it should be `getRecord*If*IndexConfigurationIsValid`... *when* refers to time, *if* is conditional.